pkg/dogstatsd/server.go

```diff
@@ -466,6 +490,87 @@ func (s *Server) storeMetricStats(sample metrics.MetricSample) {
 	ms.Name = sample.Name
 	ms.Tags = strings.Join(sample.Tags, " ") // we don't want/need to share the underlying array
 	s.Debug.Stats[key] = ms
+
+	s.Debug.metricsCounts.metricChan <- struct{}{}
+}
+
+// EnableMetricsStats enables the debug mode of the DogStatsD server and start
+// the debug mainloop collecting the amount of metrics received.
+func (s *Server) EnableMetricsStats() {
+	s.Debug.Lock()
+	defer s.Debug.Unlock()
+
+	// already enabled?
+	if atomic.LoadUint64(&s.Debug.Enabled) == 1 {
+		return
+	}
+
+	atomic.StoreUint64(&s.Debug.Enabled, 1)
+	go func() {
+		ticker := time.NewTicker(time.Millisecond * 100)
+		var closed bool
+		log.Debug("Starting the DogStatsD debug loop.")
+		for {
+			select {
+			case <-ticker.C:
+				sec := time.Now().Truncate(time.Second)
+				if sec.After(s.Debug.metricsCounts.currentSec) {
+					s.Debug.metricsCounts.currentSec = sec
+
+					if s.hasSpike() {
+						log.Warnf("A burst of metrics has been detected by DogStatSd: here is the last 5 seconds count of metrics: %v", s.Debug.metricsCounts.counts)
+					}
+
+					s.Debug.metricsCounts.bucketIdx++
+					if s.Debug.metricsCounts.bucketIdx >= len(s.Debug.metricsCounts.counts) {
+						s.Debug.metricsCounts.bucketIdx = 0
+					}
+
+					s.Debug.metricsCounts.counts[s.Debug.metricsCounts.bucketIdx] = 0
+				}
+			case <-s.Debug.metricsCounts.metricChan:
+				s.Debug.metricsCounts.counts[s.Debug.metricsCounts.bucketIdx]++
+			case <-s.Debug.metricsCounts.closeChan:
+				closed = true
+				break
+			}
+
+			if closed {
+				break
+			}
+		}
+		log.Debug("Stopping the DogStatsD debug loop.")
+		ticker.Stop()
+	}()
+}
+
+func (s *Server) hasSpike() bool {
```

Comment:
I've decided to not address this special case to avoid extra-complexity, WDYT?
It will happen with the first metrics received by DogStatsD but I don't want to ignore 0s, you still want to detect spike when DogStatsD was not receiving anything at all. Open to feedback because the current situation is not ideal.

Reply:
I do not have a strong opinion on that. I would like to make sure you did it on purpose.